test/e2e/es_index_cleaner_test.go

```diff
@@ -71,19 +77,47 @@ func esIndexCleanerTest(t *testing.T, f *framework.Framework, testCtx *framework
 	}
 	defer portForw.Close()
 	defer close(closeChan)
-	err = SmokeTest("http://localhost:16686/api/traces", "http://localhost:14268/api/traces", "foo-bar", retryInterval, timeout)
+
+	esPod, err := GetPod("default", "elasticsearch", "elasticsearch", f.KubeClient)
 	if err != nil {
 		return err
 	}
-
-	err = WaitForCronJob(t, f.KubeClient, namespace, fmt.Sprintf("%s-es-index-cleaner", name), retryInterval, timeout)
+	portForwES, closeChanES, err := CreatePortForward(esPod.Namespace, esPod.Name, []string{"9200"}, f.KubeConfig)
 	if err != nil {
 		return err
 	}
+	defer portForwES.Close()
+	defer close(closeChanES)
 
-	err = WaitForJobOfAnOwner(t, f.KubeClient, namespace, fmt.Sprintf("%s-es-index-cleaner", name), retryInterval, timeout)
+	err = SmokeTest("http://localhost:16686/api/traces", "http://localhost:14268/api/traces", "foo-bar", retryInterval, timeout)
 	if err != nil {
 		return err
 	}
-	return nil
+
+	if flag, err := hasIndex(t); !flag || err != nil {
+		return fmt.Errorf("jaeger-span index not found prior to es-index-cleaner: err = %v", err)
+	}
+
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		flag, err := hasIndex(t)
+		return !flag, err
+	})
+}
+
+func hasIndex(t *testing.T) (bool, error) {
```

Comment:
why passing `t`? maybe it's  better to change the signature to `bool containsIndex(string) bool, error`

Reply:
Yes, `t` no longer necessary so will remove. Will also update the signature and check for `jaeger-` for now.